### DISCUSSION TITLE
Allow round(num) to take a "number of digits" argument

### DISCUSSION THREAD

**Discussion by Kaedenn** [gnembon/fabric-carpet#886] [Ideas]
A scarpet implementation would be something like
`round(num) -> round(num, 0);`
`round(num, places) -> (offset = 10^places; round(num*offset)/offset);`

I'm finding myself needing this function more often than I expected and so I wrote `roundp(num, places)` to accomplish this.

**Comment by altrisi** [comment]
That's available in the default math library (built into Carpet), you can import it placing
```py
import('math', '_round');
```
at the top of your script and then call it using `_round(num, precision)`. (precision being 10, 100, 0.1, 0.001, etc)

**Reply by Kaedenn** [reply]
Wait, there are standard libraries?

**Reply by altrisi** [reply]
Yes, there's math (few distance utils) and shapes (getting block positions for a given shape with given properties). You can find their code in the repo [here](https://github.com/gnembon/fabric-carpet/tree/master/src/main/resources/assets/carpet/scripts) (the ones with the `.scl` extension).

Those should probably be documented though...
